mani_skill/vector/wrappers/gymnasium.py

```diff
@@ -141,12 +142,13 @@ def step(
         dones = torch.logical_or(terminations, truncations)
 
         if dones.any() and self.auto_reset:
-            final_obs = obs
+            final_obs = copy.deepcopy(obs)
             env_idx = torch.arange(0, self.num_envs, device=self.device)[dones]
-            obs, _ = self.reset(options=dict(env_idx=env_idx))
-            infos["final_info"] = infos.copy()
+            final_info = copy.deepcopy(infos)
```

Comment:
we should replace the deepcopy operation with one designed for torch to prefer using torch.clone

maybe a util called torch_clone_dict in the utils/common.py file?

Reply:
`infos` is a dict not a tensor, are you suggesting that we iterate over the dict and perform `.clone` if `isinstance(x, torch.Tenosr)` ?

Reply:
yes

info is a dict but may have tensor elements

Reply:
Sorry for the delay, just pushed your suggestion.